src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -437,6 +437,18 @@ sealed class BitPat(val value: BigInt, val mask: BigInt, val width: Int)
     }
   }
 
+  /** Are any bits of this BitPat `?` */
+  private[chisel3] def hasDontCares: Boolean = width > 0 && mask != ((BigInt(1) << width) - 1)
```

Comment:
width of bp cannot be 0. Maybe assert out?

Reply:
I'm future proofing this such that if we do allow it, these should be correct for it (see stubs in tests).